build/Makefile

```diff
@@ -36,11 +36,11 @@ INC=$(wildcard *.h) $(APP_ROOT)/include/*.h
 FIRMWARE_VERSION=$(shell git rev-parse HEAD)
 
 # Build this package
-$(APP_NAME).bin: $(SRC) $(INC)
+$(APP_NAME).bin: Makefile $(SRC) $(INC)
```

Comment:
I forget what the standard is with Makefiles - where does it look for this file? In the current working directory? Does it need to have a path?

Reply:
It's a bit convoluted, but here's the logic:

The ```build/Makefile``` you are looking at can only independently build StarterWare.

In order to build a rig binary, the ```build/Makefile``` *has* to be included from a rig ```Makefile```, otherwise  ```$(APP_NAME)``` is undefined and ```$(SRC)```, and ```$(INC)``` will not find the rig source code.

The ``rigs/plain/Makefile```, for example, is 

```
APP_NAME=plain
APP_ROOT=../../
SAMPLE_FREQ=1000 # 24000000/SAMPLE_FREQ must be an integer
include $(APP_ROOT)/build/Makefile
```

So, in answer to your question, when the rig ```Makefile``` is run to build a rig, it includes the ```build/Makefile``` and from within the rig ```Makefile``` the current directory is where the rig ```Makefile``` is located. The rig ```Makefile``` sets ```$(APP_NAME)``` and as ```build/Makefile``` is being included, the wildcards in it correctly find the rig source files:

```
SRC=$(wildcard *.c) $(APP_ROOT)/src/*.c
INC=$(wildcard *.h) $(APP_ROOT)/include/*.h
```

Since the wildcards run relative to the current directory, that is where the rig ```Makefile``` is, there is no need for path to the rig ```Makefile``` in the list of dependencies.

Try it! Build a rig, then change the ```SAMPLE_FREQ``` in the Makefile and run ```make``` again.